edgelet/edgelet-http/src/util/incoming.rs

```diff
@@ -70,45 +71,50 @@ impl Stream for Incoming {
                     .lock()
                     .expect("Unable to lock the connections mutex");
 
-                // Look through the connections list for the first connection that is either ready to be
-                // passed to the stream selector or has failed and needs the error bubbled up.
-                // Return a tuple containing the index and state.
-                let val = connections
-                    .iter_mut()
-                    .map(|(fut, _)| fut.poll())
-                    .enumerate()
-                    .find(|(_, result)| match result {
-                        Ok(v) => v.is_ready(),
-                        Err(_) => true,
-                    });
+                loop {
```

Comment:
You made this a loop?

Reply:
Yes. This loop covers the situation when we have several items in a `vec` that are ready to process. Imagine, if one of them has an error, and another doesn't have. We process the first one and return `Async::NotReady`. But we have another one that has no task to signal that future is ready again. 
Other words, once we started processing inside the `poll` method we should process all futures that are ready to poll.